kernel/msg_q.c

```diff
@@ -121,7 +121,8 @@ int k_msgq_cleanup(struct k_msgq *msgq)
 }
 
 
-int z_impl_k_msgq_put(struct k_msgq *msgq, const void *data, k_timeout_t timeout)
+static int z_impl_k_msgq_put_internal(struct k_msgq *msgq, const void *data,
```

Comment:
you should remove the  `z_impl_` prefix here, this should only be used for syscalls.